src/pages/postgraphile/running-postgraphile-in-docker.md

```diff
@@ -0,0 +1,598 @@
+
+## Create Docker Compose File
+
+Docker command lines can be verbose with a lot of parameters so we will use Docker Compose to orchestrate the execution of our containers. Create a new file `docker-compose.yml` at the root of the repository with the following content.
+
+```yml
+version: "3.3"
+services:
+    db:
+        container_name: forum-example-db
+        restart: always
+        image: forum-example-db
+        build:
+            context: ./db
+        volumes:
+            - db:/var/lib/postgresql/data
+        env_file:
+            - ./.env
+        networks:
+            - network
+        ports:
+            - 5432:5432
+
+networks:
+    network:
+
+volumes:
+    db:
+```
+
+### Parameters description
+
+<table>
+    <tr>
+        <th>Parameter</th><th>Description</th>
+    </tr>
+    <tr>
+        <td><b>db</b></td>
+        <td>Names of the services run by Docker Compose.</td>
+    </tr>
+    <tr>
+        <td><b>container_name</b></td>
+        <td>Guess what? It's the container name!</td>
+    </tr>
+    <tr>
+        <td><b>image</b></td>
+        <td>Name of the image to use to run the container.</td>
+    </tr>
+    <tr>
+        <td><b>build</b></td>
+        <td>When a build context is provided, Docker Compose will build a custom image using the Dockerfile located in the context folder.</td>
+    </tr>
+    <tr>
+        <td><b>context</b></td>
+        <td>Indicates the folder where to find the Dockerfile to build the image.</td>
+    </tr>
+    <tr>
+        <td><b>volumes</td>
+        <td>Mapping between the Docker volume and the PostgreSQL folder in your container, in format <b>docker_volume:container_folder</b>.<br><br>All the files generated in the <b>container_folder</b> will be copied in the <b>docker_volume</b> so that you can preserve and retrieve your data when stopping/restarting the container.<br><br>The Docker volume is automatically created when running the db container for the first time.</td>
+    </tr>
+    <tr>
+        <td><b>env_file</td>
+        <td>Path to the configuration file containing environment variables for the container. See <b>Create Configuration File</b> above.</td>
+    </tr>
+    <tr>
+        <td><b>networks</td>
+        <td>Networks are used to group and connect containers as part of a same network.</td>
+    </tr>
+    <tr>
+        <td><b>ports</td>
+        <td>Port, mapping between the port of your host machine and the port of your container, in format <b>host_port:container_port</b>.</td>
+    </tr>
+    <tr>
+        <td><b>command</td>
+        <td>Command to be executed after the container starts. Each argument must be provided in a separate list item.</td>
+    </tr>
+</table>

```

Comment:
Markdown has a table format; it's a little awkward, but perhaps easier to edit than HTML? Note: the column lines don't have to line up (but if you use `prettier` it will automatically line them up for you).

```md
| Parameter | Description |
|-----------|-------------|
| **db**  | Names of the services run by Docker Compose |
| **container_name** | Guess what? It's the container name! |
```

| Parameter | Description |
|-----------|-------------|
| **db**  | Names of the services run by Docker Compose |
| **container_name** | Guess what? It's the container name! |

Reply:
Ok, doing it.